File: LearningLens2025/legacy_files/focused-ai-grader-service/focusedai-platform/frontend/lib/core/models/test_case.dart
class TestCase {
  final String id;
  final String name;
  final String input;
  final String expectedOutput;
  final double points;
  final bool isVisible;
  final String? description;
  final int? timeoutMs;

  TestCase({
    required this.id,
    required this.name,
    required this.input,
    required this.expectedOutput,
    required this.points,
    this.isVisible = true,
    this.description,
    this.timeoutMs,
  });

  factory TestCase.fromJson(Map<String, dynamic> json) {
    return TestCase(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      input: json['input'] ?? '',
      expectedOutput: json['expectedOutput'] ?? '',
      points: (json['points'] ?? 1.0).toDouble(),
      isVisible: json['isVisible'] ?? true,
      description: json['description'],
      timeoutMs: json['timeoutMs'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'input': input,
      'expectedOutput': expectedOutput,
      'points': points,
      'isVisible': isVisible,
      if (description != null) 'description': description,
      if (timeoutMs != null) 'timeoutMs': timeoutMs,
    };
  }

  TestCase copyWith({
    String? id,
    String? name,
    String? input,
    String? expectedOutput,
    double? points,
    bool? isVisible,
    String? description,
    int? timeoutMs,
  }) {
    return TestCase(
      id: id ?? this.id,
      name: name ?? this.name,
      input: input ?? this.input,
      expectedOutput: expectedOutput ?? this.expectedOutput,
      points: points ?? this.points,
      isVisible: isVisible ?? this.isVisible,
      description: description ?? this.description,
      timeoutMs: timeoutMs ?? this.timeoutMs,
    );
  }
}

File: LearningLens2025/legacy_files/focused-ai-grader-service/focusedai-platform/backend/src/main/java/com/focusedai/model/testcase/TestCase.java
package com.focusedai.model.testcase;

public class TestCase {
    private String id;
    private String assignmentId;
    private String name;
    private String description;
    private String input;
    private String expectedOutput;
    private double points;
    private boolean isVisible = true;
    private boolean isRequired = true;
    private int timeoutMs = 30000;
    private String category = "FUNCTIONAL"; // FUNCTIONAL, PERFORMANCE, EDGE_CASE
    
    // Constructors
    public TestCase() {}
    
    public TestCase(String name, String input, String expectedOutput, double points) {
        this.name = name;
        this.input = input;
        this.expectedOutput = expectedOutput;
        this.points = points;
    }
    
    // Getters and setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    
    public String getAssignmentId() { return assignmentId; }
    public void setAssignmentId(String assignmentId) { this.assignmentId = assignmentId; }
    
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    
    public String getInput() { return input; }
    public void setInput(String input) { this.input = input; }
    
    public String getExpectedOutput() { return expectedOutput; }
    public void setExpectedOutput(String expectedOutput) { this.expectedOutput = expectedOutput; }
    
    public double getPoints() { return points; }
    public void setPoints(double points) { this.points = points; }
    
    public boolean isVisible() { return isVisible; }
    public void setVisible(boolean visible) { isVisible = visible; }
    
    public boolean isRequired() { return isRequired; }
    public void setRequired(boolean required) { isRequired = required; }
    
    public int getTimeoutMs() { return timeoutMs; }
    public void setTimeoutMs(int timeoutMs) { this.timeoutMs = timeoutMs; }
    
    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }
}